python/lsst/ts/observatory/control/base_tcs.py

```diff
@@ -273,38 +260,75 @@ async def slew_icrs(
-        elif rottype == RotType.Physical_Sky:
+        rot_frame = self.RotFrame.TARGET
+
+        # compute rotator physical position if rot_angle is sky.
+        rot_phys_val = salobj.angle_wrap_center(
+            Angle(
+                Angle(180.0, unit=u.deg)
+                + par_angle
+                - rot_angle
+                - (
+                    alt_az.alt
+                    if self.instrument_focus == InstrumentFocus.Nasmyth
+                    else 0.0
+                ),
+                unit=u.deg,
+            )
+        )
 
-            self.log.debug(f"Setting rotator physical position to {rot}.")
+        if rot_type == RotType.Sky:
+            self.log.debug(f"RotSky = {rot_angle}, RotPhys = {rot_phys_val}.")
+        elif rot_type == RotType.SkyAuto:
+            self.log.debug(f"Auto sky angle: {rot_angle}")
+            if not (self.rotator_limits[0] < rot_phys_val.deg < self.rotator_limits[1]):
+                self.log.debug(
+                    f"Rotator angle out of limits {rot_angle} [{self.rotator_limits}]. Wrapping."
+                )
+                rot_angle = salobj.angle_wrap_center(
+                    Angle(180.0, unit=u.deg) + rot_angle
+                )
+        elif rot_type == RotType.PhysicalSky:
+            self.log.debug(
+                f"Setting rotator physical position to {rot_angle}. Rotator will track sky."
+            )
 
-            rot += par_angle + alt_az.alt
+            rot_angle = rot_phys_val
 
             self.log.debug(
-                f"Parallactic angle: {par_angle.deg} | " f"Sky Angle: {rot.deg}"
+                f"Parallactic angle: {par_angle.deg} | Sky Angle: {rot_angle.deg}"
             )
-        elif rottype == RotType.Parallactic:
+        elif rot_type == RotType.Parallactic:
 
             self.log.debug(
-                f"Setting rotator position with respect to parallactic angle to {rot}."
+                f"Setting rotator position with respect to parallactic angle to {rot_angle}."
             )
 
-            rot += par_angle + 2 * alt_az.alt - 90.0
+            rot_angle = rot_phys_val + alt_az.alt - 90.0 * u.deg
 
             self.log.debug(
-                f"Parallactic angle: {par_angle.deg} | " f"Sky Angle: {rot.deg}"
+                f"Parallactic angle: {par_angle.deg} | " f"Sky Angle: {rot_angle.deg}"
+            )
+        elif rot_type == RotType.Physical:
+            self.log.debug(
+                f"Setting rotator to physical fixed position {rot_angle}. Rotator will not track."
             )
+            # FIXME: This strategy is not supported the way we would like by
+            # the pointing. We have to set the rot_angle in sky angle instead
+            # of physical value. Will check with pointing vendors to fix this.
+            # (DM-26457).
+            rot_angle = rot_phys_val
+            rot_frame = self.RotFrame.FIXED
         else:
             valid_rottypes = [f"{rt!r}" for rt in RotType].__str__()
```

Comment:
Consider `", ".join(repr(rt) for rt in RotType)` for clarity

(Also the final `__str__()` in your code wasn't doing anything because f"...{thing}..." does take take the str of the thing).